feat(arch-linux): init

ninetailedtori opened the pull request:
Closes #1870, closes #1381.

## 🎉 Theme for Arch Linux 🎉

<!--
You should give a short description of the website that you have themed.
E.g. YouTube is a video sharing platform that allows users to upload, view, and share videos.

You should also attach some screenshots of the themed website, show it off!
-->

Honestly, this is a major refactor of the existing arch-wiki theme into an umbrella "Arch Linux" userstyle, including:
- arch-wiki existing theme
  - [x] migrated to mediawiki variables?
- an additional one started by someone else for aur.archlinux.org: #1870 
  - [x] merged with this PR yet?
  - [x] finished?
- and all the others I'll be actively doing for Arch Linux!
  - [x] archlinux (home)?
  - [x] archlinux/download?
  - [x] man.archlinux?
  - [x] archlinux/packages?
  - [x] security.archlinux?
  - [x] bbs.archlinux?

## 💬 Additional Comments 💬

<!--
Include any difficulties you had theming this port, or any general comments that would be useful for the reviewer to know.
Feel free to leave this section empty if you don't have anything more to say.
-->

This is based off of the old PR that died, #1381, but will be moving the wiki to use MediaWiki variables in the library module e.g. #1895, once we can confirm it working as it should on arch's wiki (no leftover overriding on their end). All other sites will be done by me, aside from the AUR one which was done by someone else (though it'll be edited by me as well, to fall in line with the new style of formatting).

**Base: #1866**

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://userstyles.catppuccin.com/contributing/creating-userstyles/).
- [x] I have made a new directory underneath `/styles/archlinux` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml) file with information about the new userstyle.
- [x] I have included the following files:
  - [x] `catppuccin.user.less` - all the CSS for the userstyle, based on the template.

uncenter committed:
feat(lib): add mediawiki lib module

uncenter committed:
fix(lib/mediawiki): remove site specific styling

uncenter committed:
refactor(lib/mediawiki)!: nest utilities under namespace

uncenter committed:
fix(lib/mediawiki): adjust border colors slightly

uncenter committed:
fix(lib/mediawiki): adjust borders further

ninetailedtori committed:
Initial commit for manpages.
Also fix up arch wiki guidance.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Fix issues.
Now I just need to sort out the fmt issues. We can't split regexp onto multiple lines or it breaks the usercss in stylus, which is unfortunate.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

This fixes a few things - firstly, this style only works on Arch's manpage. Perhaps it's worth generating its own category, and including ALL manpage supports, but we could actually do it all under one thing.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Let's do this with a major rework.
Let's add support to all manpages to this one file, to keep it neat and organised.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Rework - let's do a rework to include it all under an archlinux category!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Rework wiki theme to fit mediawiki. This requires: #1866. manpage for arch has been finalized and perfected.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Remove README (this is autogenerated, no need to move.)

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Initial merge of PR #1870.
Being edited by me, don't mind the state of it rn, feel free to report bugs to me!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Cleanup!
Simplify some of the css, reformat.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Whoops, typo!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Move generics into global domain changes.
Fix unnecessary redefinitions.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Push a few changes to table colouring. Accidentally typo'd a change.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Big rework electric booga2
Move label, fix some bugs of overwriting, still needs some table and heading definitions to move as well. I *should've* fixed most things, but oh well.
Also, border-color: surface0 -> surface2 now!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Veeeeeeery big update, working on EVERYTHING!!!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

S...spaces? spaces???

Signed-off-by: Toria <ninetailedtori@uwu.gal>

deno fmt

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Fix a few things:
timestamp, rss-icon, colors of h3. Need to still do selective rss-icon colours.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Fix rss-icon theming.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Rework of some of the colours
Especially links css structure. Still tryna work out some bugs, oddly enough bbs overwrites a ton.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Add usercontent. Final commit for tonight!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Checklist and rbox support.
I know, should be in bed rn qwq

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Quick fix to links, quick change to li a for bbs.
Add colours to sticky and closed labels as well.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Fix link backgrounds borking on bbs.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Add support for downloads.archlinux.
Just needed h5!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Add errorlist for search errors. Fix selections.
Can't seem to fix cursor-based focus selections though. Needs debugging.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Remove cursor setting - we shouldn't touch it in our userstyles.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Remove padding from button.
Aesthetic choice, should push these upstream.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Exclude gitlab.archlinux.org in global
Issue was domain(archlinux.org) scope, now uses negative lookahead regex to solve issues.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Whoops, missed the commit.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Fixing a dicy issue with regex, everything works now!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Fix regex to ACTUALLY work!
Apparently you can't numeric modifier before lookaheads and behinds because of various logic reasons...wish you could skip when doing lookaheads, like look 2 ahead/behind etc...

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Comment changes, form input spec.
comment: // -> /**/
form input: button text color -> link color (form inputs follow link button style.)

Signed-off-by: Toria <ninetailedtori@uwu.gal>

Working on an initial change.
This won't make sense just yet.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter commented on styles/archlinux/catppuccin.user.less:
```diff
@@ -0,0 +1,386 @@
+      &logo {
+        @svg: escape(
+          '<svg xmlns="http://www.w3.org/2000/svg" version="1" width="600" height="126"><path d="M159.568 34.427c-8.89-.014-16.267 1.809-19.12 2.803l-2.937 15.857c-.007.058 14.617-3.9 21.059-3.667 10.665.383 11.646 4.076 11.46 9.06.182.292-2.752-4.503-11.979-4.664-11.64-.2-28.069 4.122-28.046 21.692-.314 19.765 14.764 25.579 25.032 25.686 9.232-.168 13.563-3.496 15.934-5.28 3.115-3.257 6.679-6.53 ...
+        );
```
This logo appears to be the exact same as on line 136 for the manpages. We can probably consolidate these? Perhaps a base `@-moz-document domain("archlinux.org")` that applies to all the subdomain pages?

ninetailedtori commented on styles/archlinux/catppuccin.user.less:
```diff
@@ -0,0 +1,386 @@
+    textarea {
+      &::placeholder {
+        color: @subtext0 !important;
+      }
```
Thanks for catching that one! I think it should be cleaned up in the next commit o7

ninetailedtori commented:
PR is now up to every page aside from:
- Download
- Wiki (needs mediawiki lib, will set up my own localhost tomorrow)
Feel free to test it for me! Security and BBS were the last ones I just did, and rework of AUR as well as expansion of it to include Packages (official) was done earlier as well! :]

uncenter commented on styles/archlinux/catppuccin.user.less:
```diff
@@ -246,9 +246,31 @@
+
+    input[type="reset"] {
+      background-color: @red !important;
+      padding: 0 2px !important;
```
Is this necessary?

ninetailedtori replied:
Yes, this is necessary so the button doesn't look cramped, unfortunately. Is it ok to keep?

uncenter reviewed (commented):
Looks like there is still a bunch of duplicated input styles?

uncenter commented on styles/archlinux/catppuccin.user.less:
```diff
@@ -246,9 +246,31 @@
+      &:hover {
+        background-color: @surface1 !important;
+        border-color: @surface2 !important;
+        cursor: pointer !important;
```
Yes please! Our policy is to not change non-color related styling properties.

ninetailedtori committed:
Fix mediawiki + remove redundants.
remove container from mw for now, add vector-pinned-container settings to inherit directly on the page (flows a lot better, fitting in line with our mw style.)

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori commented:
This is a reminder for me to change the mediawiki location once that's published, because I'm *going* to forget :P

WalkQuackBack commented:
This may have rebased this onto the wrong branch. Could you try rebasing onto origin/main again?

ninetailedtori commented:
Shoot, tried to merge but not rebase into mediawiki ;-;

WalkQuackBack committed:
Merge branch 'main' into feat/mediawiki-lib-module

ask4fusora committed:
fix(google-gemini): chat input background color (#2041)

* Add background color for the chat input

* Update version number in Google Gemini - Catppuccin user style

* chore(google-gemini): revert version metadata change

* fix(google-gemini): theme surface-bright instead of directly targeting chatbox

---------

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

github-actions[bot] committed:
chore: bump changed userstyles

ninetailedtori committed:
Do NOT break this again.
Moving stuff to lib/mediawiki to better organise miniserve in the meantime. Remove unnecessary note now that issue has been patched.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter commented on styles/archlinux/catppuccin.user.less:
```diff
@@ -0,0 +1,1269 @@
+
+    --bg-color: @base  !important;
+    --color: @text  !important;
+    --code-inline-bg-color: @mantle  !important;
```
Probably undocumented but I think we usually do surface0 for inline code blocks.

```suggestion
    --code-inline-bg-color: @surface0  !important;
```

uncenter reviewed (commented):
There's a lot of duplication between the first @-moz-document and for bbs.archlinux.com. Can we avoid that? Does the regex for the first one not include bbs?

ninetailedtori commented:
> There's a lot of duplication between the first @-moz-document and for bbs.archlinux.com. Can we avoid that? Does the regex for the first one not include bbs?

The regex DOES apply to bbs, it just isn't important enough for some reason.

I tried, but where I redefined, it wasn't applying at all. I might actually define all of these commons in a modular way, then just call the function within the required scopes at this point. Not sure why they had something overwrite it though, very annoying. I'll see what I can do though, through specificity tinkering.

ninetailedtori commented on styles/archlinux/catppuccin.user.less:
```diff
@@ -0,0 +1,1269 @@
+@import "https://userstyles.catppuccin.com/lib/lib.less";
+@import "http://127.0.0.1:8000/lib/mediawiki.less";
+
+/* deno-fmt-ignore */
```
It splits the regexp onto multiple lines, which unfortunately breaks stylus extension rn.

ninetailedtori commented on styles/archlinux/catppuccin.user.less:
```diff
@@ -0,0 +1,1269 @@
+          '<svg xmlns="http://www.w3.org/2000/svg" version="1" width="600" height="126"><path d="M159.568 34.427c-8.89-.014-16.267 1.809-19.12 2.803l-2.937 15.857c-.007.058 14.617-3.9 21.059-3.667 10.665.383 11.646 4.076 11.46 9.06.182.292-2.752-4.503-11.979-4.664-11.64-.2-28.069 4.122-28.046 21.692-.314 19.765 14.764 25.579 25.032 25.686 9.232-.168 13.563-3.496 15.934-5.28 3.115-3.257 6.679-6.53 ...
+        );
+        background: url("data:image/svg+xml,@{svg}")  !important;
+        background-size: 190px 40px  !important;
```
<img width="619" height="132" alt="image" src="https://github.com/user-attachments/assets/c74f112b-1697-444a-aaa8-d38a5b94db87" />

This is what current code looks like, i.e. size is fixed by our setting manually.

ninetailedtori committed:
Change inline code bg to surface0. Add in unimplemented values. Comment EVERYTHING in wiki section. Refactor archlinux -> arch-linux.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Remove manual setting of vector-pinned-container. The issue is both archwiki-toc-* variables should be respected upstream, but also that vector-pinned-containers should be set in mediawiki as well if possible.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter commented:
> I might actually define all of these commons in a modular way, then just call the function within the required scopes at this point.

This is what I would recommend doing.

> Not sure why they had something overwrite it though, very annoying. I'll see what I can do though, through specificity tinkering.

You could do a `&, .pun` wrapper for everything in the first moz-document and that would do the trick I think, a little ugly though.

ninetailedtori committed:
Err, add theming for git diff table.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Reformat!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Simplify regex.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter commented on styles/arch-linux/catppuccin.user.less:
```diff
@@ -0,0 +1,1347 @@
+}
+
+/* deno-fmt-ignore */
+@-moz-document regexp("^https:\\/\\/(?!gitlab\\.)(\\w+\\.)*archlinux\\.org\\/(?:packages[\\/]?)\\w*"), domain("aur.archlinux.org") {
```
Are there any subdomains of packages or does this work the same?
```suggestion
@-moz-document url-prefix("https://archlinux.org/packages"), domain("aur.archlinux.org") {
```

ninetailedtori committed:
swap to url-prefix part 1

Co-authored-by: uncenter <uncenter@uncenter.dev>

ninetailedtori committed:
part 2 catppuccin.user.less

Co-authored-by: uncenter <uncenter@uncenter.dev>

uncenter commented on styles/arch-linux/catppuccin.user.less:
```diff
@@ -0,0 +1,1347 @@
+      &logo {
+        @svg: escape('<svg xmlns="http://www.w3.org/2000/svg" version="1" width="600" height="126"><path d="M159.568 34.427c-8.89-.014-16.267 1.809-19.12 2.803l-2.937 15.857c-.007.058 14.617-3.9 21.059-3.667 10.665.383 11.646 4.076 11.46 9.06.182.292-2.752-4.503-11.979-4.664-11.64-.2-28.069 4.122-28.046 21.692-.314 19.765 14.764 25.579 25.032 25.686 9.232-.168 13.563-3.496 15.934-5.28 3.115-3.257 ...
+        background: url("data:image/svg+xml,@{svg}") !important;
+        background-size: 190px 40px !important;
```
```suggestion
        background-image: url("data:image/svg+xml,@{svg}") !important;
```

ninetailedtori replied:
Not possible to remove size, it breaks the image.

ninetailedtori commented:
> > I might actually define all of these commons in a modular way, then just call the function within the required scopes at this point.
> 
> 
> 
> This is what I would recommend doing.
> 
> 
> 
> > Not sure why they had something overwrite it though, very annoying. I'll see what I can do though, through specificity tinkering.
> 
> 
> 
> You could do a `&, .pun` wrapper for everything in the first moz-document and that would do the trick I think, a little ugly though.

I'm going to do the wrapper, tragically there's no solution. I tried refactoring to remove dupes outright and it broke as like last time I tested. I'll wrap the whole thing I think.

ninetailedtori commented on styles/arch-linux/catppuccin.user.less:
```diff
@@ -0,0 +1,1292 @@
+}
+
+/* deno-fmt-ignore */
+@-moz-document regexp("^https:\\/\\/archlinux\\.org\\/?[-a-zA-Z0-9@:%_+.~#?&\\/=]*"), domain("aur.archlinux.org") {
```
I think so? Might be wrong, lemme check. I'll add comments in the next commit!

ninetailedtori committed:
swap to background-image. catppuccin.user.less

Co-authored-by: uncenter <uncenter@uncenter.dev>

uncenter commented:
> I'm going to do the wrapper, tragically there's no solution. I tried refactoring to remove dupes outright and it broke as like last time I tested. I'll wrap the whole thing I think.

Did you try making mixins and then just applying those in the right places? I see no reason why that wouldn't work if done correctly.

ninetailedtori committed:
Typo fixes on the latest regex issue.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter commented:
Fwiw here is a good playground that I like to use: https://regex101.com/r/wrKYCM/1

uncenter commented:
> Unfortunately it seems that `\w` is broken in stylus, which means we DO have to use `.`

Yeah there isn't any reason we need to use \w over `.`. We aren't parsing or validating much here.

uncenter commented:
> That's what I've been using, have been for years! xD

Just saying I put the test strings for each of the URLs in there so it's easier.

ninetailedtori committed:
svg changes. restructuring into mixins will be next.
note: security.archlinux.org or aur.archlinux.org icons might not work if content-security-policy blocks it.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Revert to a crazy fix attempt, remove dupe, use background-image solution.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Quickly move around some stuff in brdmenu. Fix up stats. Gotta go fast! Gotta do mixins next.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori commented:
> > That's what I've been using, have been for years! xD
> 
> Just saying I put the test strings for each of the URLs in there so it's easier.

Ya, that's what I've been doing! :3 it is annoying though, I have to then test it twice, because stylus regex isn't fully with the same accepted tokens as it is in that tester, so I have to check every time what sites are recognised. xD anyway rn I'm doing mixins, so will be big update soon!

uncenter committed:
fix(go.dev): cve banner & icons (#2061)

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
chore: update generated files

ninetailedtori committed:
Big batch of mixins and mayhem!
TOC variables should NOT be used, and possibly the other archwiki variables in the wiki, but am awaiting response on whether this will change: https://gitlab.archlinux.org/archlinux/archwiki/-/issues/91#note_387552

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Perfect the quotebox highlighting on hover!
I might be actually godly at this now...knock on wood...

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Quick url patch, accidentally auto-formatted an error!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add FIXME comment, gotta work on this next.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add dl, dt, dd. This covers newslist on main menu. Welcoming testing/fixing on this, hover box AND alternating colours don't seem to match up with this yet unfortunately.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

TheAnonymousCrusher committed:
feat(poe): init (#2069)

* Add files via upload

* Update styles/poe/catppuccin.user.less

Change 'poe' to 'Poe'

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

* Update styles/poe/catppuccin.user.less

Change 'poe' to 'Pow'

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

* Update scripts/userstyles.yml

Co-authored-by: uncenter <uncenter@uncenter.dev>

* Update scripts/userstyles.yml

Co-authored-by: uncenter <uncenter@uncenter.dev>

* Remove variable that are defined in standard lib

Co-authored-by: uncenter <uncenter@uncenter.dev>

* Fix indentation in scripts/userstyles.yml

* fix(poe): indentation

* chore(poe): delete extra file

* Update styles/poe/catppuccin.user.less

Co-authored-by: uncenter <uncenter@uncenter.dev>

* Update styles/poe/catppuccin.user.less

Co-authored-by: uncenter <uncenter@uncenter.dev>

* Update catppuccin.user.less

* Remove prism.js like syntax

* Change to CSS variables

Co-authored-by: uncenter <uncenter@uncenter.dev>

* fix(poe): settings, tooltips, codes, misc

* Update styles/poe/catppuccin.user.less

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

* Update catppuccin.user.less

* Change 'Tooltip_root' BG to surface0 to show elevation

* fix(poe): use variable instead

* feat(poe): logo, ignore help center, api section

* fix(poe): optimize logo svg

* fix(poe): login gradient, add comment about url-prefix

---------

Co-authored-by: WalkQuackBack <quacksareback@duck.com>
Co-authored-by: uncenter <uncenter@uncenter.dev>

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
chore: update generated files

github-actions[bot] committed:
style: deno fmt

ninetailedtori committed:
Add footer, add ssdiff, add clarity fixes to colours on diff viewing on package sites. Need a second opinion on the background highlighting, might need to make it even more transparent?

Signed-off-by: Toria <ninetailedtori@uwu.gal>

r58Playz committed:
chore(formative): remove `r58Playz` as maintainer (#2073)

github-actions[bot] committed:
chore: update generated files

koibtw committed:
chore(userstyles.yml): `adamperkowski` -> `koibtw` (#2074)

github-actions[bot] committed:
chore: update generated files

ninetailedtori committed:
Fix some inputs
1. add mw-ui-icon-wiki-expand definition to remove our colours.
2. move borders to accent
3. add focus-within tags to all inputs, to theme accurately.
4. move enabled, that shouldn't even be there, should be with &:enabled!
    - I need to still do disabled for all inputs though.
5. Add input[type="submit"], which is useful for some pages, e.g.
    - https://wiki.archlinux.org/index.php?title=Special:Search&profile=images&search=&fulltext=1

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Fix a few styling consistency things with deno task lint.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

WalkQuackBack committed:
chore(startpage): add `WalkQuackBack` as maintainer (#2071)

github-actions[bot] committed:
chore: update generated files

WalkQuackBack committed:
feat(lib/mediawiki): use codex style guide (#1962)

* feat(lib/mediawiki): use codex style guide to base colour choices

* chore(lib/mediawiki): comment nits

* fix(lib/mediawiki): change button quiet hover colours to fade accent

* fix(mediawiki): remove redundant transparent variables

* fix(lib/mediawiki): change tab list framed state colours to latte fade

---------

Co-authored-by: uncenter <uncenter@uncenter.dev>

ninetailedtori committed:
Working on radio and checkbox. Still need to add disabled.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Fix the search, but it's still broken.
Test disabled entry? Uncertain if it works yet.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter committed:
feat(lib/mediawiki): radio inputs

ninetailedtori committed:
Push radio widget theming upstream.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Merge branch 'feat/mediawiki-lib-module' into manpages

koibtw committed:
feat(pronouns.page): add language-specific domains (#2075)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter commented on styles/arch-linux/catppuccin.user.less:
```diff
@@ -0,0 +1,1501 @@
+        border-color: @blue transparent transparent transparent !important;
+      }
+    }
+  }
```
I don't understand the styling here. This doesn't match the original site.

<img width="1164" height="730" alt="Image" src="https://github.com/user-attachments/assets/fa84fa4a-244a-47c1-b737-a2cb5836648b" />

<img width="1312" height="856" alt="Image" src="https://github.com/user-attachments/assets/b9a66b2c-b107-4ac4-a1dc-994974213426" />

This could just be this, according to the styling of the original site:

```css
      a, a:link, a:visted {
        color: @accent !important;
      }
```

ninetailedtori committed:
Commit radio input changes!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori commented on styles/arch-linux/catppuccin.user.less:
```diff
@@ -0,0 +1,1501 @@
+    // FIXME: This doesn't work yet, but I want to just apply filter to all span icons :]
+    span[background-image="url(Air/img/bull.png)"] {
+      filter: @accent-filter !important;
+    }
```
global labels are required, because otherwise we'd have to add about 30 different places to set it. It's equivalent for all of them unless otherwise overwritten, which takes a much lower line of code. Footers are the same, every page has a footer. Legends the same, there's a lot of legends, similar to labels, used just as much. Content I can see if I can simplify, but I did try iirc, and there were a lot of things that broke.

uncenter replied:
Let's at the least avoid `content *`.

ninetailedtori committed:
Big batch of "match the og style" :] not finished yet, given I deleted a lot of my og style, and need to work it back.

ninetailedtori committed:
Generalise the inputs, to attack email correctly (can't be set selectively or we get CSP banned.)

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Push big change to some other stuff again, cleaning up.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Trying to fix bbs now.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Big rework on bbs. fixing EVERYTHING again!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Fix quick bug with blockquotes.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Adding some final fixes for bbs.

ninetailedtori committed:
Still working on stuff, mostly wiki fix. One more syntax error left?

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter committed:
feat(regex101): init (#1188)

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
chore: update generated files

github-actions[bot] committed:
style: deno fmt

ninetailedtori committed:
Tidy up archnavbarlist. Move colour to text by default.

ninetailedtori committed:
Amend formatting, formatter seems broken.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Fix the background-image changes.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ImenaOphelia committed:
feat(anonymous-overflow): init (#1564)

Co-authored-by: WalkQuackBack <quacksareback@duck.com>
Co-authored-by: uncenter <uncenter@uncenter.dev>

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
chore: update generated files

github-actions[bot] committed:
style: deno fmt

ninetailedtori committed:
Fix specificity of navbarlist.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add theming per subpage on aur.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
MORE PATCHING!!!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Comment the aur subpage units!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Fix checklists, fix some headers.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add radio inputs pr, add fix for links. Needs to add upstream too, ignore for now.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Update mediawiki url to upstream.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Fix a weird mw issue. Will push upstream.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Remove important. Seems to work fine now.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter committed:
fix(arch-linux): apply mediawiki lib module correctly

ninetailedtori committed:
Remove fix from mw.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

WalkQuackBack commented on styles/arch-linux/catppuccin.user.less:
```diff
@@ -0,0 +1,1602 @@
+==/UserStyle== */
+
+@import "https://userstyles.catppuccin.com/lib/lib.less";
+@import "http://127.0.0.1:8000/lib/mediawiki.less";
```
Reminder to change this before merging.

WalkQuackBack commented on styles/arch-wiki/catppuccin.user.less:
We should keep this around for at least a few months, with the updateURL pointing to the new userstyle, in a similar way to the CSS > Less migration. Then, current users will be able to migrate properly.

uncenter replied:
@ninetailedtori we will take care of this, don't worry about doing this yourself.

ninetailedtori committed:
Add comments header formatting, and some metadata settings. Also, nest metadata under pkgdetails for neatness.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add border change to headers. Useful in way too many places, not neatly overwritten otherwise.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add pin-comment, remove dupe.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Little too specific on the css. Mimic upstream aurweb.css, root .*-comment classes.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Fix mw thing

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
remove importance from radio inputs, add oo-ui-input and button.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Fix diffs for wiki.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Move icon inversion to mw lib.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Move .mw-protection-indicator-icon--lock to global, unneeded by bbs afaik.

ninetailedtori committed:
Add fallback for browsers that don't support mask-image protocols.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add extiw to the link icons.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add error, move to minify slightly.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add background-color inherit to fix delete-comment bug.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add override for manpages.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter committed:
feat(brave-search): rewrite (#2072)

* feat(brave-search): rewrite

* feat(brave-search): add style for settings page

* feat(brave-search): use background-image instead of background

---------

Co-authored-by: ndsboy <12872019+ndsboy@users.noreply.github.com>

github-actions[bot] committed:
chore: bump changed userstyles

ninetailedtori committed:
Comment spree, fix AUR register subdupe, write login as separate.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Make comment-header apply to base as well, required for some other comment-headers.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
fix(docs.rs): toolbar icons, sidebar border (#2087)

github-actions[bot] committed:
chore: bump changed userstyles

ninetailedtori committed:
Add some old bug page theming.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add mw-file-element. Remove importance from filters, breaks icons.
Add intl support!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Move icon theming to base. Add check for svg. Arch only uses simple icons.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Push change, add specificity, add icon.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Formatting manually. Linter broken.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

incognitotgt committed:
feat(scalar): init (#1628)

* feat: init

* formatted places i shouldnt touch

* add antoher domain

* chore(scalar): delete preview

* chore(scalar): `readme.app-link` to `link`

* style(scalar): newlines

* fix(scalar): update supportURL metadata

* fix(scalar): misc

* refactor(scalar): use standard lib

* Update styles/scalar/catppuccin.user.less

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

* feat(scalar): home & dashboard pages, syntax highlighting, primary buttons, logo and misc

* fix(scalar): add brightness lifted filter override back

Whoops, found it now.

* fix(scalar): set color purple to mauve instead of lavender

* fix(scalar): use base/surface bg scale over base/mantle/crust

---------

Co-authored-by: uncenter <uncenter@uncenter.dev>
Co-authored-by: WalkQuackBack <quacksareback@duck.com>

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
chore: update generated files

github-actions[bot] committed:
style: deno fmt

ninetailedtori committed:
Add merge-2 icon.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori commented on styles/arch-linux/catppuccin.user.less:
```diff
@@ -0,0 +1,1931 @@
+==/UserStyle== */
+
+@import "https://userstyles.catppuccin.com/lib/lib.less";
+@import "http://127.0.0.1:8000/catppuccin-styles/userstyles/lib/mediawiki.less";
```
```suggestion
@import "https://userstyles.catppuccin.com/lib/mediawiki.less";
```

ninetailedtori committed:
Move mediawiki to eventual upstream.

ninetailedtori committed:
Optimise svg.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Remove silly things, add next and prev.

ninetailedtori committed:
Add refresh red arrows.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
optimise svg again.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Codebox fixes.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

WalkQuackBack committed:
feat(lib/lib): use more accurate COBYLA CSS filters (#2106)

feat(lib/lib): COBYLA CSS filters

Updates the Catppuccin Filters in the standard library to use the COBYLA CSS filters in https://github.com/catppuccin/whiskers/pull/116

ninetailedtori committed:
Fix links for visited etc.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
set default postsig colour.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Push some specificity about the tables.
Add arch-bio-entry formatting.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Whoops, wack a reset on that header include again.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter committed:
fix(chess.com): outside board coordinates (#2121)

github-actions[bot] committed:
chore: bump changed userstyles

ninetailedtori committed:
Add mailing list support, add manpages correction, add fix for brdmenu isactive tab.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
add a deno fmt. task lint throwing a fit where it shouldn't, ignore.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Further support for mailing lists site.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Work to add manpages mdoc macro definitions. Consider moving to lib if popular enough, like most manpage sites use it.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

42willow committed:
fix(duckduckgo): add duck.ai domain (#2101)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
feat(lib/lib.less): add rgbify/hslify config options (#2018)

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

uncenter committed:
docs: remove arguments from mixin section headings (#2130)

uncenter committed:
revert: "feat(lib/lib.less): add rgbify/hslify config options (#2018)" (#2133)

This reverts commit 168f796a82565be5d8f9d7547d624524fcb482a2.

ninetailedtori committed:
Push some changes for the blob table, nest nest nest!!!

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Push some deno fixes to fix the linting throws.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Push some more blocking.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Push fix for thead and th.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

a3chron committed:
fix(github): feed link color (#2136)

Co-authored-by: uncenter <uncenter@uncenter.dev>

github-actions[bot] committed:
chore: bump changed userstyles

ninetailedtori committed:
Push bootstrap import, needs changing when bs lib gets made.
Need to make a pf lib too.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

uncenter committed:
fix(youtube): hide sampled tint (#2138)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
fix(youtube): video hover preview and shorts progress bars (#2139)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
fix(vercel): prismjs syntax highlighting (#2144)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
chore(ISSUE_TEMPLATE): add form fields for stylus and userstyle versions (#2127)

Rewords grammar to be more concise and adds addition fields to assist with reproducing userstyle bugs.

* chore(ISSUE_TEMPLATE): add form fields for stylus and userstyle versions

* chore: apply suggestions for wording & non-q labels

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

* fix(ISSUE_TEMPLATE): add (s) to browser version description

Co-authored-by: uncenter <uncenter@uncenter.dev>

* chore: add description for stylus version

* chore: catppuccin userstyle version label

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

* chore: replace tab with spaces

---------

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

github-actions[bot] committed:
chore: update generated files

koibtw committed:
feat(searxng): apply filter to logo image (#2148)

github-actions[bot] committed:
chore: bump changed userstyles

ddosian committed:
fix(whatsapp-web): modal-backdrop-solid isn't themed (#2147)

themed modal-backdrop-solid

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
fix(chess.com): appearance mode selectors (#2145)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
fix(chess.com): primary buttons (#2120)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
fix(web.dev): exclude subdomains (#2150)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
feat(chatgpt): rewrite (#2059)

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
style: deno fmt

ionawr committed:
fix(codeberg): header elements (#2108)

Co-authored-by: uncenter <uncenter@uncenter.dev>

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
style: deno fmt

rohanp2051 committed:
fix(reddit): trending carousel cards text (#2114)

Co-authored-by: uncenter <uncenter@uncenter.dev>

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
style: deno fmt

uncenter committed:
fix(codeberg): navbar profile dropdown text (#2154)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
refactor(reddit): remove unused rules and variables (#2155)

github-actions[bot] committed:
chore: bump changed userstyles

kripikripi committed:
fix(reddit): background color behind subreddit name (#1915)

Co-authored-by: Alex <kripikripi09@gamil.com>
Co-authored-by: uncenter <uncenter@uncenter.dev>

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
style: deno fmt

uncenter committed:
fix(ghostty): brand buttons and active sidebar, homepage terminal (#2160)

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
style: deno fmt

uncenter committed:
feat(mdn): rewrite (#2157)

Co-authored-by: Kyosuke Kobayashi <kyo0117dragonquest@gmail.com>

uncenter committed:
fix(nixos-search): misc (#2156)

github-actions[bot] committed:
chore: bump changed userstyles

scarcekoi committed:
fix(claude): misc (#2163)

Co-authored-by: uncenter <uncenter@uncenter.dev>

github-actions[bot] committed:
chore: bump changed userstyles

renovate[bot] committed:
chore(deps): update dependency svgo to v4 (#1790)

* chore(deps): update dependency svgo to v4

* chore: regenerate `deno.lock`

* chore(stylelint/optimized-svgs): update svgo syntax for v4.x

* refactor: apply svg lint fixes

Looks like the only difference is that the  order of functions in the
`transform` attribute was changed, shouldn't have any functional
differences.

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: WalkQuackBack <quacksareback@duck.com>
Co-authored-by: uncenter <uncenter@uncenter.dev>

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
fix(duckduckgo): lavender visited links (#2068)

github-actions[bot] committed:
chore: bump changed userstyles

jn-sena committed:
chore(mullvad-leta)!: remove (#2180)

jn-sena committed:
chore(hyperpipe): remove `jn-sena` as maintainer (#2181)

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
chore: update generated files

renovate[bot] committed:
chore(deps): update actions/labeler action to v6 (#1883)

renovate[bot] committed:
chore(deps): update dependency @catppuccin/starlight to v1.1.1 (#2167)

uncenter committed:
chore: revert "chore(deps): update actions/labeler action to v6 (#1883)" (#2184)

uncenter committed:
chore(github): note themes/contrast settings in usage note (#2197)

github-actions[bot] committed:
chore: update generated files

pynappo committed:
fix(canvas-lms): support bettercampus over bettercanvas (#2200)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
fix(youtube): notification bell and tint override (#2196)

github-actions[bot] committed:
chore: bump changed userstyles

scarcekoi committed:
fix(mdbook): add blockquote, quiz support and match upstream (#2190)

Co-authored-by: uncenter <uncenter@uncenter.dev>

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
fix(claude): higher specificity applicator (#2205)

github-actions[bot] committed:
chore: bump changed userstyles

scarcekoi committed:
feat(claude): add missing css variables (#2164)

github-actions[bot] committed:
chore: bump changed userstyles

uncenter committed:
chore(PULL_REQUEST_TEMPLATE/userstyle-update): revamp (#2203)

uncenter committed:
chore(ISSUE_TEMPLATE): various improvements (#2189)

github-actions[bot] committed:
chore: update generated files

uncenter committed:
feat(chess.com): neo piece set & always use mocha colors (#2151)

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
style: deno fmt

scarcekoi committed:
feat(modrinth): add missing css variables and update platform colors (#2202)

Co-authored-by: WalkQuackBack <quacksareback@duck.com>

github-actions[bot] committed:
chore: bump changed userstyles

42willow committed:
feat(npmx): init (#2176)

Co-authored-by: uncenter <uncenter@uncenter.dev>

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
chore: update generated files

WalkQuackBack committed:
docs(lib/mediawiki): initial documentation writeup (#2089)

uncenter committed:
fix(youtube): misc (#2207)

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
style: deno fmt

uncenter committed:
fix(chatgpt): sidebar, settings, tooltips (#2211)

github-actions[bot] committed:
chore: bump changed userstyles

github-actions[bot] committed:
style: deno fmt

ninetailedtori committed:
Merge remote-tracking branch 'upstream/main' into manpages

ninetailedtori committed:
Add some work for accounts.archlinux.org.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add avatar icon too.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Add loading container

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Revert "Merge remote-tracking branch 'upstream/main' into manpages"

This reverts commit 9a46870367fff969b179f9080ee3599a0367e1ab, reversing
changes made to 94ca77efa69573ef08f8a6b80a0ff51d022b49e3.

ninetailedtori committed:
Merge branch 'feat/mediawiki-lib-module' into manpages

ninetailedtori committed:
Add important as red.

Signed-off-by: Toria <ninetailedtori@uwu.gal>

ninetailedtori committed:
Merge remote-tracking branch 'origin/manpages' into manpages
